InnerEye/ML/model_training.py

```diff
@@ -71,6 +71,19 @@ def write_args_file(config: Any, outputs_folder: Path) -> None:
     logging.info(output)
 
 
+class InnerEyeRecoveryCheckpointCallback(ModelCheckpoint):
+    def __init__(self, container: LightningContainer):
+        super().__init__(dirpath=str(container.checkpoint_folder),
+                         monitor="epoch",
+                         filename=RECOVERY_CHECKPOINT_FILE_NAME + "_{epoch}",
+                         period=container.recovery_checkpoint_save_interval,
+                         save_top_k=container.save_last_k_recovery_checkpoints,
+                         mode="max")
+
+    def on_train_epoch_end(self, trainer: Trainer, pl_module: LightningModule, outputs: Any) -> None:
+        pl_module.log(name="epoch", value=trainer.current_epoch)
```

Comment:
Why is this happening?

Reply:
Because if we don't log epoch, we can not keep the last k checkpoints. PL lightning requires to monitor something (here epoch) when save_top_k is set to something different than None (i.e. saving more than the last).